modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala

```diff
@@ -536,18 +541,25 @@ final case class SharedOptions(
   def compilerMaker(
     threads: BuildThreads,
     scaladoc: Boolean = false
-  ): Either[BuildException, ScalaCompilerMaker] = either {
+  ): Either[BuildException, ScalaCompilerMaker] =
     if (scaladoc)
-      SimpleScalaCompilerMaker("java", Nil, scaladoc = true)
+      Right(SimpleScalaCompilerMaker("java", Nil, scaladoc = true))
     else if (compilationServer.server.getOrElse(true))
-      new BloopCompilerMaker(
-        value(bloopRifleConfig()),
-        threads.bloop,
-        strictBloopJsonCheckOrDefault
-      )
+      bloopRifleConfig() match {
+        case Right(config) =>
+          Right(new BloopCompilerMaker(
+            config,
+            threads.bloop,
+            strictBloopJsonCheckOrDefault,
+            coursier.getOffline().getOrElse(false)
+          ))
+        case Left(ex) if coursier.getOffline().contains(true) =>
+          logger.debug("Couldn't fetch a JVM for Bloop from cache, fallback to using scalac")
```

Comment:
Similarly, I'd promote this to a warning.
```suggestion
          logger.message("Warning: JVM for Bloop could not be fetched from the local cache; offline mode is on, so using scalac as fallback")
```